compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -529,12 +529,12 @@ hlintSettings hlintDataDir = do
     findSettings (unmask . readSettingsFile (Just hlintDataDir)) Nothing
   return (classify, hints)
 
-getHlintSettingsRule :: Maybe HlintUsage -> Rules ()
+getHlintSettingsRule :: HlintUsage -> Rules ()
 getHlintSettingsRule usage =
     defineNoFile $ \GetHlintSettings ->
       liftIO $ case usage of
-          Just (HlintEnabled dir) -> hlintSettings dir
-          _ -> fail "linter configuration unspecified"
+          HlintEnabled dir -> hlintSettings dir
+          HlintDisabled -> fail "linter configuration unspecified"
```

Comment:
Is this code actually reachable?

Reply:
It is reachable if you request it the rule explicitly e.g. in a test.

Reply:
Would it make sense to push the pattern match into the caller of this rule?

Reply:
No by virtue of the `hintEnabled` flag in the `ofInterestRule`.

Reply:
If so, then I guess that would be here:
```
getHlintDiagnosticsRule :: Rules ()
getHlintDiagnosticsRule =
    define $ \GetHlintDiagnostics file -> do
        pm <- use_ GetParsedModule file
        let anns = pm_annotations pm
        let modu = pm_parsed_source pm
        (classify, hint) <- useNoFile_ GetHlintSettings -- <- here!
        let ideas = applyHints classify hint [createModuleEx anns modu]
        return ([diagnostic file i | i <- ideas, ideaSeverity i /= Ignore], Just ())
    where
```
The regular code path elides calls to `getHlintDiagnosticsRule` via the `hlintEnabled` flag in the `ofInterestRule`:
```
        let hlintEnabled = case optHlintUsage opts of
              HlintEnabled _ -> True
              HlintDisabled -> False
        ...
        let hlintActions = [use_ GetHlintDiagnostics f | hlintEnabled, f <- files]
```
Let me know what you want to do.

Reply:
^ I'm inclined to land this. Will raise a new PR if @martin-drhu-da  deems changes desirable.

Reply:
I had another look. I misunderstood it the previous time. I'm happy to leave it as is. Thanks.